system/jlib/jstring.hpp

```diff
@@ -407,7 +407,7 @@ extern jlib_decl void decodeXML(ISimpleReadStream &in, StringBuffer &out, unsign
 extern jlib_decl int utf8CharLen(unsigned char ch);
 extern jlib_decl int utf8CharLen(const unsigned char *ch, unsigned maxsize = (unsigned)-1);
 
-extern jlib_decl StringBuffer &replaceString(StringBuffer & result, size_t lenSource, const char *source, size_t lenOldStr, const char* oldStr, size_t lenNewStr, const char* newStr);
+extern jlib_decl bool replaceString(StringBuffer & result, size_t lenSource, const char *source, size_t lenOldStr, const char* oldStr, size_t lenNewStr, const char* newStr, bool forceCopy = false);
```

Comment:
In general I prefer not to use default values for parameters, they tend to cause more problems than they solve.
In this case if there was a default, it should default the opposite way, because seeing to false is actually an optimization that changes the natural expectation of the implementation of the function.

If I am being picky then "forceCopy" isn't a great parameter name.  Better is something like "copyIfUnmatched", or possibly better "avoidCopyIfUnmatched" with the opposite meaning - so it is a parameter that you only set to true if you want the special behaviour.

Reply:
Removed the default value and changed the parameter name to "avoidCopyIfUnmatched".